--- fvhlib-2.5/Makefile.orig	Fri Feb 21 21:15:22 2003
+++ fvhlib-2.5/Makefile	Wed May 28 17:15:21 2003
@@ -19,7 +19,7 @@
 CC=gcc
 
 CFLAGS=-fPIC -Wshadow -Wwrite-strings -Wconversion
-LDFLAGS=-L/usr/local/contrib/lib -L/usr/local/lib -lm
+LDFLAGS=-L/usr/local/contrib/lib -L/usr/local/lib -lm -lssl
 VERSION=2.5
 
 OBJS=io.o proc.o ssl.o str.o utils.o val.o tcp.o snd_dev.o snd_util.o ip.o mail.o mem.o main.o
@@ -28,7 +28,7 @@
 
 lib: $(OBJS)
 	ar -r libfvh.a $(OBJS)
-	ld -x -shared -soname libfvh.so.2 -o libfvh.so.$(VERSION) --whole-archive libfvh.a
+	$(CC) -shared -Wl,-soname=libfvh.so.2 $(LDFLAGS) -o libfvh.so.$(VERSION) $(OBJS)
 
 install: all
 	cp libfvh.* /usr/local/lib
